routes/characters.js

```diff
@@ -1,133 +1,33 @@
 var express = require('express');
 var router = new express.Router();
 var mongoose = require('mongoose');
-
+var CharacterService = require('../services/characterService');
 var Character = require('../models/Character');
 
-router.get('/', (req, res, next) => {
-  Character.find()
-    .select('_id name level')
-    .exec()
-    .then((docs) => {
-      var response = {
-        count: docs.length,
-        characters: docs.map((doc) => {
-          return {
-            _id: doc._id,
-            name: doc.name,
-            level: doc.level,
-            request: {
-              type: 'GET',
-              url: 'http://localhost:3000/characters/' + doc._id
-            }
-          };
-        })
-      };
-      res.status(200).json(response);
-    })
-    .catch((err) => {
-      console.log(err);
-      res.status(500).json({ error: err });
-    });
-});
+const characterServiceInstance = new CharacterService(Character,mongoose);
```

Comment:
We can just `const mongoose = require('mongoose')` and the model `var Character = require('../models/Character')` inside the service then we don't have to intantiate it with them :)
Then mongoose can also be removed from the route files, separating concerns ;)

Reply:
was that not the point of dependency injection?

Reply:
It's mainly the dependency to the service we need to inject ;)
I also think we need some kind of service container, but we can add that in the future :)

Reply:
Okay, this was what I found as examples when reading up on dependency injection with express and nodeJS

Reply:
I see he uses it for the models too in the article we talked about - nevermind ^^
But at least we don't need to inject mongoose (as we're only using it for id generation anyway.

Reply:
I am not sure what you mean (or want), but if you have a correction you are welcome to make it ;)

Reply:
I believe what @mschier wants is the following
`const characterServiceInstance = new CharacterService(Character,mongoose);-->const characterServiceInstance = new CharacterService(Character);` And then add mongoose in CharacterService as it isnt needed in the route

Reply:
Indeed (I changed this in my newest commits). It turns out that the only thing we were even using mongoose for was ObjectId generation, which can be done automatically in the models - so no need for the mongoose dependency after all :)